docs/source/about_dataset_features.rst

```diff
@@ -48,4 +48,16 @@ The ``answers`` field is constructed using the :class:`datasets.Sequence` featur
 
 .. tip::
 
-   See the :ref:`flatten` section to learn how you can extract the nested subfields as their own independent columns.
\ No newline at end of file
+   See the :ref:`flatten` section to learn how you can extract the nested subfields as their own independent columns.
+
+The :class:`datasets.ArrayXD` feature is useful for creating arrays of various sizes. You can create arrays with two dimensions using :class:`datasets.Array2D`, and even arrays with five dimensions using :class:`datasets.Array5D`. 
+
+.. code::
+
+   >>> features = Features({'a': Array2D(shape=(1, 3), dtype='int32'))
+
+:class:`datasets.ArrayXD` also allows the first dimension of the array to be dynamic. This is useful for handling sequences with variable lengths such as sentences, without having to pad or truncate the input to a uniform shape.
```

Comment:
I think `datasets.ArrayXD` doesn't exist. Also I think in the long run we want to call it `ArrayND` instead - N is better for integers.

```suggestion
The array type also allows the first dimension of the array to be dynamic. This is useful for handling sequences with variable lengths such as sentences, without having to pad or truncate the input to a uniform shape.
```